truetype/truetype.go

```diff
@@ -497,7 +497,8 @@ func (f *Font) Index(x rune) Index {
 		return Index(val)
 	}
 
-	if len(f.charcodeToGID) > int(c) {
+	if len(f.charcodeToGID) > 0 && c < 256 {
```

Comment:
do we need to make sure uint8(c) is within the size of the slice?  like that it doesn't go out of bounds?

Reply:
This was my initial logic but this is a map so we can have less entries but still have necessary codes in it. 
The regression I had with a couple of files was when map contains for example 50 entries but still has entry for c=200.

Reply:
ok, i was thinking more of a perspective of avoiding it from crashing if trying to access something out of bounds

Reply:
ok, if its a map it's probably fine, i guess if its not in the map we just get a 0 value